src/server/zset_family.cc

```diff
@@ -2599,6 +2636,296 @@ void ZSetFamily::GeoDist(CmdArgList args, ConnectionContext* cntx) {
+          return (*cntx)->SendError(kInvalidFloatErr);
+        }
+        string_view unit;
+        unit = ArgS(args, i + 2);
+        shape.conversion = ExtractUnit(unit);
+        if (shape.conversion == -1) {
+          return (*cntx)->SendError("unsupported unit provided. please use M, KM, FT, MI");
+        }
+        shape.type = CIRCULAR_TYPE;
+        by_set = true;
+        i += 2;
+      } else {
+        return (*cntx)->SendError(kSyntaxErr);
+      }
+    } else if (cur_arg == "BYBOX") {
+      if (by_set) {
+        return (*cntx)->SendError(kByRadiusBoxErr);
+      } else if (i + 3 < args.size()) {
+        if (!ParseDouble(ArgS(args, i + 1), &shape.t.r.width)) {
+          return (*cntx)->SendError(kInvalidFloatErr);
+        }
+        if (!ParseDouble(ArgS(args, i + 2), &shape.t.r.height)) {
+          return (*cntx)->SendError(kInvalidFloatErr);
+        }
+        string_view unit;
+        unit = ArgS(args, i + 3);
+        shape.conversion = ExtractUnit(unit);
+        if (shape.conversion == -1) {
+          return (*cntx)->SendError("unsupported unit provided. please use M, KM, FT, MI");
+        }
+        shape.type = RECTANGLE_TYPE;
+        by_set = true;
+        i += 3;
+      } else {
+        return (*cntx)->SendError(kSyntaxErr);
+      }
+    } else if (cur_arg == "ASC") {
+      if (desc) {
+        return (*cntx)->SendError(kAscDescErr);
+      } else {
+        asc = true;
+      }
+    } else if (cur_arg == "DESC") {
+      if (asc) {
+        return (*cntx)->SendError(kAscDescErr);
+      } else {
+        desc = true;
+      }
+    } else if (cur_arg == "COUNT") {
+      if (i + 1 < args.size()) {
+        // TODO: need a string_view impl of stoull
+        count = std::stoull(std::string(ArgS(args, i + 1)));
+        i++;
+      } else {
+        return (*cntx)->SendError(kSyntaxErr);
+      }
+      if (i + 1 < args.size() && ArgS(args, i + 1) == "ANY") {
+        any = true;
+        i++;
+      }
+    } else if (cur_arg == "WITHCOORD") {
+      withcoord = true;
+    } else if (cur_arg == "WITHDIST") {
+      withdist = true;
+    } else if (cur_arg == "WITHHASH")
+      withhash = true;
+    else {
+      return (*cntx)->SendError(kSyntaxErr);
+      break;
+    }
+  }
+
+  // check mandatory options
+  if (!from_set) {
+    return (*cntx)->SendError(kSyntaxErr);
+  }
+  if (!by_set) {
+    return (*cntx)->SendError(kSyntaxErr);
+  }
+
+  // query
+  GeoHashRadius georadius = geohashCalculateAreasByShapeWGS84(&shape);
+  GeoArray ga;
+  membersOfAllNeighbors(cntx, key, &georadius, &shape, ga, any ? count : 0);
+
+  // if no matching results, the user gets an empty reply.
+  if (ga.empty()) {
+    (*cntx)->SendNull();
+  }
+
+  // sort and trim by count
+  auto asc_comparator = [](const GeoPoint& a, const GeoPoint& b) { return a.dist < b.dist; };
+  auto desc_comparator = [](const GeoPoint& a, const GeoPoint& b) { return a.dist > b.dist; };
```

Comment:
Can you reduce the scope of `asc_comparator` and `desc_comparator` to the smallest possible?